crates/router_env/src/logger/formatter.rs

```diff
@@ -174,20 +179,17 @@ where
         #[cfg(feature = "vergen")]
         let build = crate::build!().to_string();
         let env = crate::env::which().to_string();
-        default_fields.retain(|key, value| {
-            if !IMPLICIT_KEYS.contains(key.as_str()) {
-                true
-            } else {
-                tracing::warn!(
-                    ?key,
-                    ?value,
-                    "Attempting to log a reserved entry. It won't be added to the logs"
-                );
-                false
+        for key in default_fields.keys() {
+            if IMPLICIT_KEYS.contains(key.as_str()) {
+                return Err(ConfigError::Message(format!(
+                    "Invalid configuration value for key: {}",
```

Comment:
Can you update the log message?
And please include a screenshot of _this_ log message being printed.